ai21/clients/common/assistant/assistant.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List
+
+from ai21.models.responses.assistant_response import AssistantResponse
+from ai21.types import NotGiven, NOT_GIVEN
+from ai21.utils.typing import remove_not_given
+
+
+class Assistant(ABC):
+    _module_name = "assistants"
+
+    @abstractmethod
+    def create(
+        self,
+        name: str,
+        *,
+        description: str | NotGiven = NOT_GIVEN,
+        optimization: str | NotGiven = NOT_GIVEN,
+        avatar: str | NotGiven = NOT_GIVEN,
+        models: List[str] | NotGiven = NOT_GIVEN,
+        tools: List[str] | NotGiven = NOT_GIVEN,
+        tool_resources: dict | NotGiven = NOT_GIVEN,
+        **kwargs,
+    ) -> AssistantResponse:
+        """
+        :param name: The name of the assistant.
+        :param description: The description of the assistant.
+        :param optimization: The optimization to use.
+        :param avatar: The avatar to use.
+        :param models: The models to use.
+        :param tools: The tools to use.
+        :param tool_resources: The tool resources to use.
+        :param kwargs: Additional keyword arguments.
+        :return: The response object.
```

Comment:
Doc-string is not really needed IMO if the description is saying the same thing as the name suggests